src/processor/operator/ddl/create_table.cpp

```diff
@@ -12,19 +13,41 @@ namespace processor {
 void CreateTable::executeDDLInternal(ExecutionContext* context) {
     auto clientContext = context->clientContext;
     auto catalog = clientContext->getCatalog();
-    switch (info.onConflict) {
-    case common::ConflictAction::ON_CONFLICT_DO_NOTHING: {
-        if (catalog->containsTable(clientContext->getTransaction(), info.tableName)) {
+    auto transaction = clientContext->getTransaction();
+    // Check conflict
+    // We don't allow naming conflicting between tables and rel groups because rel groups are
+    // transparent to user.
+    auto contains = catalog->containsTable(transaction, info.tableName) ||
+                    catalog->containsRelGroup(transaction, info.tableName);
+    if (contains) {
+        switch (info.onConflict) {
+        case ConflictAction::ON_CONFLICT_DO_NOTHING: {
             return;
         }
+        case ConflictAction::ON_CONFLICT_THROW: {
+            throw BinderException(info.tableName + " already exists in catalog.");
+        }
+        default:
+            KU_UNREACHABLE;
+        }
     }
+    // Create table.
+    CatalogEntry* entry = nullptr;
+    switch (info.type) {
```

Comment:
I wonder if we can move the switch case into `Catalog::createTableEntry()`, and remove `createNodeTableEntry/createRelTableEntry/createRelGroupEntry` from `Catalog` public interfaces.

Reply:
Certainly we can merge createNode/RelTableEntry.

But Relgroup needs to be its own because every CatalogSet should have its own create function.